src/lib.rs

```diff
@@ -124,25 +125,29 @@ impl<L: Launcher> TsunamiBuilder<L> {
     /// method.
     ///
     /// The default duration is 6 hours.
-    pub fn set_max_duration(&mut self, hours: u64) {
+    pub fn set_max_duration(&mut self, hours: u64) -> &mut Self {
         self.max_duration = Some(time::Duration::from_secs(hours * 3600));
+        self
     }
 
     /// Set the logging target for this tsunami.
     ///
     /// By default, logging is disabled (i.e., the default logger is `slog::Discard`).
-    pub fn set_logger(&mut self, log: slog::Logger) {
+    pub fn set_logger(&mut self, log: slog::Logger) -> &mut Self {
         self.log = log;
+        self
     }
 
     /// Enable logging to terminal.
-    pub fn use_term_logger(&mut self) {
+    pub fn use_term_logger(&mut self) -> &mut Self {
         use slog::Drain;
         use std::sync::Mutex;
 
         let decorator = slog_term::TermDecorator::new().build();
         let drain = Mutex::new(slog_term::FullFormat::new(decorator).build()).fuse();
         self.log = slog::Logger::root(drain, o!());
+
+        self
```

Comment:
With this change I think we should make `spawn` take `&self` so that you can use the same builder multiple times if you wish.

Reply:
See also https://rust-lang-nursery.github.io/api-guidelines/type-safety.html#c-builder

Reply:
because some of the MachineDescriptors are not `Clone`, it needs to be `&mut self`, with the descriptor list being drained each time.

Reply:
Why can't `MachineDescriptor` be `Clone`?

Reply:
The descriptors hold a setup function (https://github.com/jonhoo/tsunami/pull/22/files/23e32b6cc39d9a9280c19d016b724e3b9f3bc9a1#diff-60699702750aef5a8cdc050b4fce64eaR58), which was a useful feature in the old API. 
These don't impl Clone?

Reply:
Hmm, yeah, that's a little awkward. It _is_ possible to bound the function by `Clone`, but you're right that not doing so is _probably_ better.

Reply:
Although actually, if all you need in `Fn` (i.e., you don't take `FnMut`), then you can just store `Arc` of the setup function, no?

Reply:
Ah, good idea re: `Arc`, it works for the provider implementations we have now.
I could imagine future cases where users might want `FnMut` setup functions, but that can be left to future work.